server/static/page.jsx

```diff
@@ -1,8 +1,404 @@
+          <td>{ this.props.device.addon_version }</td>
+        </tr>
+        <tr>
+          <th>session count</th>
+          <td>{ this.props.device.session_count }</td>
+        </tr>
+        <tr>
+          <th>number of shots</th>
+          <td>{ this.props.device.shotCount }</td>
+        </tr>
+        { rows }
+      </tbody>
+    </table>;
+  }
+
+  onClickLookUp(shot, event) {
+    document.querySelector("#lookup-shot-input").value = shot.id;
+    let form = document.querySelector("#lookup-shot-form");
+    form.querySelector("button").click();
+    setTimeout(() => {
+      form.scrollIntoView();
+    }, 100);
+  }
+
+  onClickBlock(shot, event) {
+    let input = document.querySelector("#block-shot-input");
+    input.value = shot.id;
+    input.focus();
+  }
+
+  onBlockDevice() {
+    let input = document.querySelector("#block-device-input");
+    input.value = this.props.device.deviceId;
+    input.focus();
+  }
+}
+
+class BlockDevice extends React.Component {
+  render() {
+    return <fieldset>
+      <legend>block device</legend>
+      <form onSubmit={this.onSubmit.bind(this)}>
+        Block deviceId:
+        <input
+        placeholder="deviceId"
+        ref={input => this.input = input}
+        id="block-device-input"
+        type="text" />
+        <button type="submit">Block</button>
+      </form>
+      <hr />
+      { this.props.blockDeviceInformation ? this.props.blockDeviceInformation : null }
+    </fieldset>;
+  }
+
+  async onSubmit(event) {
+    event.preventDefault();
+    let deviceId = this.input.value;
+    model.blockDeviceInformation = "executing...";
+    render();
+    let resp = await fetch("/block-device", {
+      body: JSON.stringify({deviceId}),
+      headers: {"content-type": "application/json"},
+      method: "POST",
+    });
+    if (!resp.ok) {
+      let body = await resp.text();
+      model.lastError = {
+        description: "Error looking up device:",
+        status: resp.status,
+        statusText: resp.statusText,
+        body: body,
+      };
+      model.blockDeviceInformation = "Errored";
+      render();
+      return;
+    }
+    model.blockDeviceInformation = `Blocked deviceId ${deviceId}`;
+    render();
+  }
+}
+
+class BlockShot extends React.Component {
+  render() {
+    return <fieldset>
+      <legend>block shot</legend>
+      <form onSubmit={this.onSubmit.bind(this)}>
+        Block shot:
+        <input
+        placeholder="shot ID, shot or image URL"
+        ref={input => this.input = input}
+        id="block-shot-input"
+        type="text" />
+        <select
+        ref={select => this.select = select} required>
+          <option value="">Select block reason</option>
+          <option value="none">Clear block</option>
+          <option value="dmca">DMCA</option>
+          <option value="usererror">User error</option>
+          <option value="illegal">Illegal</option>
+          <option value="illegal-quarantine">Illegal + quarantine</option>
```

Comment:
Same question here about possible states including / not including 'watchdog'. Should we pull this list from the schema? If not, maybe file a bug on screenshots to comment in the schema about updating the admin interface when the enum changes...